### PR TITLE
TestRunner code cleanup

### PR DESCRIPTION
**Pull request by aktondak**
#### Problem solved by the commit
This code cleans up the TestRunner class by moving all the APIs which do not belong there to the TestValidateUtilities namespace.
This code clean-up is an effort towards building the runner recipe based xrt-smi validate test. This will facilitate the following:
1. Usage of moved APIs freely out of the TestRunner class hierarchy as well.
2. Making TestRunner objects lightweight since every validate test spawns a TestRunner object upon creation. Currently, TestRunner is bloated, acting as a placeholder for all test related utility APIs, which should not be the case.
3. Allow for code re-use between multiple xrt-smi validate tests and eventually Run Recipe based Test runs.

_Note_: ps+aie, ps+validate tests may be still required for embedded+platform

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
https://jira.xilinx.com/browse/VITIS-13774
Discovered to internal analysis of Validate Tests.

#### How problem was solved, alternative solutions (if any) and why they were rejected
The problem was solved by moving the mentioned APIs out of the class and altering the usage points of these APIs to have appropriate scopes.

#### Risks (if any) associated the changes in the commit
None

#### What has been tested and how, request additional testing if necessary
Tested by running all xrt-smi validate tests on krackenmachine

#### Documentation impact (if any)
None

### CODE REVIEW

**Review by AShivangi — CHANGES_REQUESTED** [review]
I think this is a good start. Some comments. 

In the next iteration, we should separate out the boiler plate code, like "init_hw_context", "init_BO" etc. All these can live in XBValidateUtils

File: src/runtime_src/core/tools/common/tests/TestBist.cpp
Comment (aktondak) [reply src/runtime_src/core/tools/common/tests/TestBist.cpp:6]: Max pointed to a confirmation from @chvamshi-xilinx 
Waiting for reply to decide on this clean-up

File: src/runtime_src/core/tools/common/tests/TestValidateUtilities.cpp
Comment (aktondak) [reply src/runtime_src/core/tools/common/tests/TestValidateUtilities.cpp:279]: Cleaned up after confirmation

File: src/runtime_src/core/tools/common/tests/TestPsIops.cpp
Comment (AShivangi) [inline src/runtime_src/core/tools/common/tests/TestPsIops.cpp:6]: Please remove this testcase as well

File: src/runtime_src/core/tools/common/tests/TestValidateUtilities.cpp
Comment (AShivangi) [reply src/runtime_src/core/tools/common/tests/TestValidateUtilities.cpp:279]: Waiting to make sure if we still support any of the legacy devices, before this PR is ready for merge

**Pull request merged**
